### PR TITLE
Handle wildcard types in generics

### LINKED ISSUE #77: Generics bug with "? extends _____"
If I have a list like so:

`List<? extends Thing>`

The generated code fails to compile because it looks like this:

`moshi.adapter(Types.newParameterizedType(List.class, ? extends Thing.class));`

### PR DESCRIPTION
**Pull request by ZacSweers**
These were getting the full "? extends/super ______" inlined before and failing to compile, this ensures they only have one bound (multiple bounds aren't representable types) and takes the one bound they declare as the parameterized type.

Resolves #77

### CODE REVIEW

File: auto-value-moshi/src/test/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtensionTest.java [hunk 1]
```diff
@@ -110,8 +110,8 @@
         + "    private final JsonAdapter<Integer> cAdapter;\n"
         + "    private final JsonAdapter<String> dAdapter;\n"
         + "    private final JsonAdapter<Map<String, Number>> eAdapter;\n"
-        + "    private final JsonAdapter<Set<String>> fAdapter;\n"
-        + "    private final JsonAdapter<Map<String, Set<String>>> gAdapter;\n"
+        + "    private final JsonAdapter<Set<? extends String>> fAdapter;\n"
```
Comment (swankjesse) [inline auto-value-moshi/src/test/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtensionTest.java:115]: Moshi’s `adapter()` method below trusts you to get your types right. But here you’re slightly cheating, coercing a `Set<String>` to a `Set<? extends String>`. In practice it’s the same, but it isn’t guaranteed in general.

Might be better to just add the `Types.subtypeOf()`.

**Comment by ZacSweers** [comment]
Will resolve conflicts in a rebase after reviews are done so I don't blow away comments

File: auto-value-moshi/src/test/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtensionTest.java (hunk 1 above)
Comment (ZacSweers) [reply auto-value-moshi/src/test/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtensionTest.java:115]: 6764269

**Review by rharter — APPROVED** [review]

**Comment by ZacSweers** [comment]
Rebase conflicts resolved

**Pull request merged**